src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -435,6 +461,26 @@ public void write() {
     }
   }
 
+  private void startRecovery() {
```

Comment:
Where are the test cases for the new code?

Reply:
Will need your inputs if any for this as well. Thanks in advance.

Reply:
The class takes a `WriterProvider`. You can use that to pass a mock `Writer` that throws an exception on close. You could make it throw an exception the first time close is called, but not the second or third time, and you could make it _always_ throw an exception, so you can test both the case where the error is recoverable, and the case where the error is not recoverable.

Likewise, you could have `WriterProvider` pass an exception-throwing writer the first time it is asked for a writer, but a non-exception-throwing writer the second time it's asked for a writer, to verify that the recover can, indeed, recover.